edi_oca/models/edi_exchange_record.py

```diff
@@ -209,7 +209,7 @@ def record(self):
             return None
         if not self.model and self.parent_id:
             return self.parent_id.record
-        return self.env[self.model].browse(self.res_id).exists()
+        return self.env[self.model].browse(self.res_id).with_user(self._uid).filtered(lambda x: x.sudo().company_id in self.env.companies.ids).exists()
```

Comment:
Better use `self.env.uid` than `self._uid`, since the latter form is deprecated.
But I am not even sure that we need `with_user` here.
```suggestion
        record = self.env[self.model].search([("id", "=", self.res_id), ("company_id", "in", self.env.companies.ids)])
        return record.with_user(self.env.uid)
```

or
```suggestion
        return self.env[self.model].search([("id", "=", self.res_id), ("company_id", "in", self.env.companies.ids)])
```